app.js

```diff
@@ -55,4 +55,19 @@ app.use(function(err, req, res, next) {
   res.render('error');
 });
 
+app.get('/users/:id', (req, res) => {
+ const userId = Number(req.params.id);
+ const user = users.find((user) => user.id === userId);
+ console.log(req.query)
+ if (!user) {
+ if (req.query.url) {
+ res.redirect(req.query.url);
```

Comment:
* **Rule ID:** javascript.express.security.audit.express-open-redirect.express-open-redirect
* **File:** app.js
* **Line:** 64
* **Description:** The application redirects to a URL specified by user-supplied input `req` that is not validated. This could redirect users to malicious locations. Consider using an allow-list approach to validate URLs, or warn users they are being redirected to a third-party website.
* **Impact:** MEDIUM
* **Confidence:** HIGH
* **Semgrep Rule:** [Link](https://semgrep.dev/r/javascript.express.security.audit.express-open-redirect.express-open-redirect)